pymeasure/instruments/keithley/keithley2182.py

```diff
@@ -0,0 +1,399 @@
+        Valid values are True and False.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+    voltage_offset = Channel.control(
+        ":SENS:VOLT:CHAN{ch}:REF?", ":SENS:VOLT:CHAN{ch}:REF %g",
+        """Control the relative offset for measuring voltage.
+        Displayed value = actual value - offset value.
+        Valid ranges are -120 V to +120 V for Ch. 1, and -12 V to +12 V for Ch. 2.""",
+        validator=strict_range,
+        values=[-120, 120],
+        dynamic=True,
+    )
+    temperature_offset = Channel.control(
+        ":SENS:TEMP:CHAN{ch}:REF?", ":SENS:TEMP:CHAN{ch}:REF %g",
+        """Control the relative offset for measuring temperature.
+        Displayed value = actual value - offset value.
+        Valid values are -273 C to 1800 C.""",
+        validator=strict_range,
+        values=[-273, 1800],
+    )
+    voltage_offset_state = Channel.control(
+        ":SENS:VOLT:CHAN{ch}:REF:STAT?", ":SENS:VOLT:CHAN{ch}:REF:STAT %s",
+        """Control whether voltage is measured as a relative or absolute value.
+        Enabled by default for Ch. 2 voltage, which is measured relative to Ch. 1
+        voltage.""",
+        validator=strict_discrete_set,
+        values={'OFF': 0, 'ON': 1},
+        map_values=True
+    )
+    temperature_offset_state = Channel.control(
+        ":SENS:TEMP:CHAN{ch}:REF:STAT?", ":SENS:TEMP:CHAN{ch}:REF:STAT %s",
+        """Control whether temperature is measured as a relative or absolute value.
+        Disabled by default.""",
+        validator=strict_discrete_set,
+        values={'OFF': 0, 'ON': 1},
+        map_values=True
+    )
+
+    def setup_voltage(self, auto_range=True, nplc=5):
+        """Set active channel and configure channel for voltage measurement.
+
+        :param auto_range: Enables auto_range if True, else uses set voltage
+            range
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 50/60
+        """
+        self.write(":SENS:CHAN {ch};"
+                   ":SENS:FUNC 'VOLT';"
+                   f":SENS:VOLT:NPLC {nplc};")
+        if auto_range:
+            self.write(":SENS:VOLT:RANG:AUTO 1")
+        self.check_errors()
+
+    def setup_temperature(self, nplc=5):
+        """Change active channel and configure channel for temperature measurement.
+
+        :param channel: Temperature measurement channel
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 50/60
+        """
+        self.write(":SENS:CHAN {ch};"
+                   ":SENS:FUNC 'TEMP';"
+                   f":SENS:TEMP:NPLC {nplc}")
+        self.check_errors()
+
+    def acquire_reference(self, func='VOLT'):
+        """Acquire a measurement and store it as the relative offset value.
+
+        Only acquires reference if offset state is ON.
+
+        :param func: 'VOLT' or 'TEMP'
+        """
+        self.write(f":SENS:{func}:""CHAN{ch}:REF:ACQ")
```

Comment:
Consider making two methods, one for voltage and one for temperature.
Using strings for choosing options is not that handy.